index.js

```diff
@@ -99,7 +94,60 @@ async function pay (plugin, {
   }
 }
 
+async function pull (plugin, {
+  pointer,
+  amount,
+  streamOpts = {},
+  callback,
```

Comment:
In contrast to push payments, where the client (usually) knows how much value she sends, she does not know how much she will be able to pull (because the token is opaque and the SPSP endpoint query response only has optional parameters). The callback is used to do anything with the actual amount pulled. For the CLI it is just printing that value, for Codius, it could be computing the expiration time. 
I guess you can argue that this would be beneficial for push payments, too, in the case when the total amount could not be sent.

Reply:
>I guess you can argue that this would be beneficial for push payments, too, in the case when the total amount could not be sent.

I'd be in favor of that

Reply:
I can add that. That should be a different PR though, right?

Reply:
:+1: Separate pr is fine